Unable to share file anymore

`zboxcli git:(staging) ./zbox share --allocation $alloc --remotepath /folder1 --clientid $clientid --encryptionpublickey $enc_key
0chain-core-sdk [INFO]   2021/10/13  wallet.go:290: ******* Wallet SDK Version:v1.2.87 *******
0box-sdk        [INFO]   2021/10/13  sdk.go:72: ******* Storage SDK Version: v1.2.87 *******
0chain-core-sdk [INFO]   2021/10/13  wallet.go:401: *******  Wallet SDK Version:v1.2.87 *******
0box-sdk        [DEBUG]  2021/10/13  networkworker.go:103: Get network result:{"miners":["https://dev.0chain.net/miner01","https://dev.0chain.net/miner03","https://dev.0chain.net/miner02"],"sharders":["https://dev.0chain.net/sharder02","https://dev.0chain.net/sharder01"]}
0box-sdk        [INFO]   2021/10/13  filemetaworker.go:73: File Meta result:{"chunk_size":0,"collaborators":[],"commit_meta_txns":[],"created_at":"2021-10-13T14:37:02.793613Z","hash":"c45f97aecdb988ada559c16bba5226176436e982f1a9a3ca246b38c2dd7d5ace","lookup_hash":"d43a0778aa1d26f46f8d8fe57bec2734a0d68c06403d517953952089e064a2dc","name":"folder1","num_of_blocks":3,"path":"/folder1","path_hash":"0e526a00d697c4beca434a54a4545cad06c3fb83d01d6ad0cd2a8a047f967dfe","size":173674,"type":"d","updated_at":"2021-10-13T14:37:02.793613Z"}
0box-sdk        [INFO]   2021/10/13  filemetaworker.go:73: File Meta result:{"chunk_size":0,"collaborators":[],"commit_meta_txns":[],"created_at":"2021-10-13T14:37:02.94158Z","hash":"c45f97aecdb988ada559c16bba5226176436e982f1a9a3ca246b38c2dd7d5ace","lookup_hash":"d43a0778aa1d26f46f8d8fe57bec2734a0d68c06403d517953952089e064a2dc","name":"folder1","num_of_blocks":3,"path":"/folder1","path_hash":"0e526a00d697c4beca434a54a4545cad06c3fb83d01d6ad0cd2a8a047f967dfe","size":173674,"type":"d","updated_at":"2021-10-13T14:37:02.94158Z"}
0box-sdk        [ERROR]  2021/10/13 allocation.go:1090: http://dev.0chain.net:31304 Insert share info error response: 400{"error":"ERROR: value too long for type character varying(64) (SQLSTATE 22001)"}


0box-sdk        [ERROR]  2021/10/13 allocation.go:1090: http://dev.0chain.net:31306 Insert share info error response: 400{"error":"ERROR: value too long for type character varying(64) (SQLSTATE 22001)"}`

--- a/code/go/0chain.net/blobbercore/handler/handler.go
+++ b/code/go/0chain.net/blobbercore/handler/handler.go
@@ -452,10 +452,6 @@ func InsertShare(ctx context.Context, r *http.Request) (interface{}, error) {
 		return nil, common.NewError("share_info_insert", "Wrong ownerID or clientID")
 	}
 
-	if allocationObj.OwnerID != authTicket.OwnerID {
-		return nil, common.NewError("share_info_insert", "Wrong owner of allocation")
-	}
-
 	shareInfo := reference.ShareInfo{
 		OwnerID:                   authTicket.OwnerID,
 		ClientID:                  authTicket.ClientID,

--- a/code/go/0chain.net/blobbercore/handler/handler.go
+++ b/code/go/0chain.net/blobbercore/handler/handler.go
@@ -447,6 +447,15 @@ func InsertShare(ctx context.Context, r *http.Request) (interface{}, error) {
 		return nil, err
 	}
 
+	// dummy, to avoid input and sql error
+	if len(authTicket.ClientID) != 64 || len(authTicket.OwnerID) != 64 {
+		return nil, common.NewError("share_info_insert", "Wrong ownerID or clientID")
+	}
+
+	if allocationObj.OwnerID != authTicket.OwnerID {
+		return nil, common.NewError("share_info_insert", "Wrong owner of allocation")
+	}
+
 	shareInfo := reference.ShareInfo{
 		OwnerID:                   authTicket.OwnerID,
 		ClientID:                  authTicket.ClientID,
@@ -464,7 +473,7 @@ func InsertShare(ctx context.Context, r *http.Request) (interface{}, error) {
 		err = reference.AddShareInfo(ctx, shareInfo)
 	}
 	if err != nil {
-		return nil, err
+		return nil, common.NewError("share_info_insert", "Unable to save share info")
 	}
 
 	resp := map[string]interface{}{

--- a/code/go/0chain.net/blobbercore/handler/handler_test.go
+++ b/code/go/0chain.net/blobbercore/handler/handler_test.go
@@ -1188,12 +1188,12 @@ func TestHandlers_Requiring_Signature(t *testing.T) {
 					)
 
 				mock.ExpectQuery(regexp.QuoteMeta(`SELECT * FROM "marketplace_share_info" WHERE`)).
-					WithArgs("abcdefgh", "f15383a1130bd2fae1e52a7a15c432269eeb7def555f1f8b9b9a28bd9611362c").
+					WithArgs("da4b54d934890aa415bb043ce1126f2e30a96faf63a4c65c25bbddcb32824d77", "f15383a1130bd2fae1e52a7a15c432269eeb7def555f1f8b9b9a28bd9611362c").
 					WillReturnRows(sqlmock.NewRows([]string{}))
 				aa := sqlmock.AnyArg()
 
 				mock.ExpectExec(`INSERT INTO "marketplace_share_info"`).
-					WithArgs(client.GetClientID(), "abcdefgh", "f15383a1130bd2fae1e52a7a15c432269eeb7def555f1f8b9b9a28bd9611362c", "regenkey", aa, false, aa).
+					WithArgs(client.GetClientID(), "da4b54d934890aa415bb043ce1126f2e30a96faf63a4c65c25bbddcb32824d77", "f15383a1130bd2fae1e52a7a15c432269eeb7def555f1f8b9b9a28bd9611362c", "regenkey", aa, false, aa).
 					WillReturnResult(sqlmock.NewResult(0, 0))
 			},
 			wantCode: http.StatusOK,
@@ -1276,7 +1276,7 @@ func TestHandlers_Requiring_Signature(t *testing.T) {
 					)
 
 				mock.ExpectQuery(regexp.QuoteMeta(`SELECT * FROM "marketplace_share_info" WHERE`)).
-					WithArgs("abcdefgh", "f15383a1130bd2fae1e52a7a15c432269eeb7def555f1f8b9b9a28bd9611362c").
+					WithArgs("da4b54d934890aa415bb043ce1126f2e30a96faf63a4c65c25bbddcb32824d77", "f15383a1130bd2fae1e52a7a15c432269eeb7def555f1f8b9b9a28bd9611362c").
 					WillReturnRows(
 						sqlmock.NewRows([]string{"client_id", "owner_id"}).
 							AddRow("abcdefgh", "owner"),
@@ -1284,7 +1284,7 @@ func TestHandlers_Requiring_Signature(t *testing.T) {
 				aa := sqlmock.AnyArg()
 
 				mock.ExpectExec(`UPDATE "marketplace_share_info"`).
-					WithArgs("regenkey", "kkk", false, aa, "abcdefgh", "f15383a1130bd2fae1e52a7a15c432269eeb7def555f1f8b9b9a28bd9611362c").
+					WithArgs("regenkey", "kkk", false, aa, "da4b54d934890aa415bb043ce1126f2e30a96faf63a4c65c25bbddcb32824d77", "f15383a1130bd2fae1e52a7a15c432269eeb7def555f1f8b9b9a28bd9611362c").
 					WillReturnResult(sqlmock.NewResult(0, 1))
 			},
 			wantCode: http.StatusOK,
@@ -1363,7 +1363,7 @@ func TestHandlers_Requiring_Signature(t *testing.T) {
 					)
 
 				mock.ExpectExec(regexp.QuoteMeta(`UPDATE "marketplace_share_info"`)).
-					WithArgs(true, "abcdefgh", filePathHash).
+					WithArgs(true, "da4b54d934890aa415bb043ce1126f2e30a96faf63a4c65c25bbddcb32824d77", filePathHash).
 					WillReturnResult(sqlmock.NewResult(0, 1))
 
 			},
@@ -1443,7 +1443,7 @@ func TestHandlers_Requiring_Signature(t *testing.T) {
 					)
 
 				mock.ExpectExec(regexp.QuoteMeta(`UPDATE "marketplace_share_info"`)).
-					WithArgs(true, "abcdefgh", filePathHash).
+					WithArgs(true, "da4b54d934890aa415bb043ce1126f2e30a96faf63a4c65c25bbddcb32824d77", filePathHash).
 					WillReturnResult(sqlmock.NewResult(0, 0))
 
 			},

--- a/code/go/0chain.net/blobbercore/handler/handler_test.go
+++ b/code/go/0chain.net/blobbercore/handler/handler_test.go
@@ -1125,7 +1125,7 @@ func TestHandlers_Requiring_Signature(t *testing.T) {
 					body := bytes.NewBuffer(nil)
 					formWriter := multipart.NewWriter(body)
 					shareClientEncryptionPublicKey := "kkk"
-					shareClientID := "abcdefgh"
+					shareClientID := "da4b54d934890aa415bb043ce1126f2e30a96faf63a4c65c25bbddcb32824d77"
 					require.NoError(t, formWriter.WriteField("encryption_public_key", shareClientEncryptionPublicKey))
 					remotePath := "/file.txt"
 					filePathHash := "f15383a1130bd2fae1e52a7a15c432269eeb7def555f1f8b9b9a28bd9611362c"
@@ -1213,7 +1213,7 @@ func TestHandlers_Requiring_Signature(t *testing.T) {
 					body := bytes.NewBuffer(nil)
 					formWriter := multipart.NewWriter(body)
 					shareClientEncryptionPublicKey := "kkk"
-					shareClientID := "abcdefgh"
+					shareClientID := "da4b54d934890aa415bb043ce1126f2e30a96faf63a4c65c25bbddcb32824d77"
 					require.NoError(t, formWriter.WriteField("encryption_public_key", shareClientEncryptionPublicKey))
 					remotePath := "/file.txt"
 					filePathHash := "f15383a1130bd2fae1e52a7a15c432269eeb7def555f1f8b9b9a28bd9611362c"
@@ -1303,7 +1303,7 @@ func TestHandlers_Requiring_Signature(t *testing.T) {
 
 					body := bytes.NewBuffer(nil)
 					formWriter := multipart.NewWriter(body)
-					shareClientID := "abcdefgh"
+					shareClientID := "da4b54d934890aa415bb043ce1126f2e30a96faf63a4c65c25bbddcb32824d77"
 					remotePath := "/file.txt"
 
 					require.NoError(t, formWriter.WriteField("refereeClientID", shareClientID))
@@ -1383,7 +1383,7 @@ func TestHandlers_Requiring_Signature(t *testing.T) {
 
 					body := bytes.NewBuffer(nil)
 					formWriter := multipart.NewWriter(body)
-					shareClientID := "abcdefgh"
+					shareClientID := "da4b54d934890aa415bb043ce1126f2e30a96faf63a4c65c25bbddcb32824d77"
 					remotePath := "/file.txt"
 
 					require.NoError(t, formWriter.WriteField("refereeClientID", shareClientID))

